api/public/public_api/endpoints/wake_word_file.py

```diff
@@ -0,0 +1,169 @@
+# Mycroft Server - Backend
+# Copyright (C) 2020 Mycroft AI Inc
+# SPDX-License-Identifier: 	AGPL-3.0-or-later
+#
+# This file is part of the Mycroft Server.
+#
+# The Mycroft Server is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+"""Public Device API endpoint for uploading a sample wake word for tagging."""
+from datetime import datetime
+from http import HTTPStatus
+from logging import getLogger
+from os import environ
+from pathlib import Path
+
+from flask import jsonify
+from schematics import Model
+from schematics.types import StringType
+from schematics.exceptions import DataError
+
+from selene.api import PublicEndpoint
+from selene.data.account import Account, AccountRepository
+from selene.data.tagging import (
+    TaggingFileLocationRepository,
+    WakeWordFile,
+    WakeWordFileRepository,
+)
+from selene.data.wake_word import WakeWordRepository
+
+LOCAL_IP = "127.0.0.1"
+
+_log = getLogger(__package__)
+
+
+class UploadRequest(Model):
+    """Data class for validating the content of the POST request."""
+
+    wake_word = StringType(required=True)
+    engine = StringType(required=True)
+    timestamp = StringType(required=True)
+    model = StringType(required=True)
+
+
+class WakeWordFileUpload(PublicEndpoint):
+    """Endpoint for submitting and retrieving wake word sample files.
+
+    Samples will be saved to a temporary location on the API host until a daily batch
+    job moves them to a permanent one.  Each file will be logged on the sample table
+    for their location and classification data.
+    """
+
+    _file_location = None
+    _wake_word_repository = None
+    _wake_word = None
+
+    def __init__(self):
+        super(WakeWordFileUpload, self).__init__()
+        self.request_data = None
+
+    @property
+    def wake_word_repository(self):
+        """Lazy instantiation of wake word repository object."""
+        if self._wake_word_repository is None:
+            self._wake_word_repository = WakeWordRepository(self.db)
```

Comment:
Where is the exception caught for this call and what to do if response is None?

Reply:
I am not sure I understand your question.  This is a pretty standard way to lazily instantiate a property.  The class variable with the preceding underscore is only None until the first time the property is referenced.  After that, only the value is returned.

Reply:
I checked and I thought the code left the value None if it failed. It seems to assume the value of self.db can never be none or that acquiring the connection can never fail so that is my question. Where is the failure caught and what will the user see when it does fail?

Reply:
self.db is a property of the base class.  it will resolve when referenced.  should never be none

Reply:
What happens if it can't connect to the database? Where/how is that handled? I don't see that.